File: org.nextframework.stjs/js-builder/org/nextframework/resource/NextSuggestStaticListProvider.java
package org.nextframework.resource;

import org.nextframework.resource.NextSuggest.SuggestElement;
import org.stjs.javascript.Array;
import org.stjs.javascript.JSCollections;


public class NextSuggestStaticListProvider extends NextSuggestSuggestionProvider {
	
	private Array<SuggestItem> items;

	public NextSuggestStaticListProvider(Array<SuggestItem> items){
		this.items = items;
	}

	@Override
	public void requestSuggestions(SuggestElement suggestElement) {
		String queryText = suggestElement.getQueryText();
		Array<SuggestItem> suggestions = JSCollections.$array();
		for (String i : items) {
			SuggestItem item = items.$get(i);
			if(suggestElement.itemMatcher.match(queryText, item)){
				suggestions.push(item);
			}
		}
		suggestElement.suggest(suggestions);
	}

}

File: org.nextframework.view/resources/org/nextframework/resource/NextSuggestStaticListProvider.js
var NextSuggestStaticListProvider = function(items) {
    NextSuggestSuggestionProvider.call(this);
    this.items = items;
};
stjs.extend(NextSuggestStaticListProvider, NextSuggestSuggestionProvider);

NextSuggestStaticListProvider.prototype.items = null;
NextSuggestStaticListProvider.prototype.requestSuggestions = function(suggestElement) {
    var queryText = suggestElement.getQueryText();
    var suggestions = [];
    for (var i in this.items) {
        if (!(this.items).hasOwnProperty(i)) continue;
        var item = this.items[i];
        if (suggestElement.itemMatcher.match(queryText, item)) {
            suggestions.push(item);
        }
    }
    suggestElement.suggest(suggestions);
};
NextSuggestStaticListProvider.$typeDescription=stjs.copyProps(NextSuggestSuggestionProvider.$typeDescription, {"items":{name:"Array", arguments:["NextSuggestSuggestionProvider.SuggestItem"]}});

